index.js

```diff
@@ -89,13 +89,28 @@ function create(win, opts) {
 
 		// filter out leading/trailing separators
 		// TODO: https://github.com/electron/electron/issues/5869
-		menuTpl = menuTpl.filter((el, i, arr) => !(el.type === 'separator' && (i === 0 || i === arr.length - 1 || arr[i + 1].type === 'separator')));
+		menuTpl = deleteUnnecessarySeparatorsAndElements(menuTpl);
 
-		const menu = (electron.Menu || electron.remote.Menu).buildFromTemplate(menuTpl);
-		menu.popup(win);
+		if (menuTpl.some(elem => isVisible(elem))) {
```

Comment:
This menu is already filtered, so couldn't you just do: `if (menuTpl.length > 0) {` ?